Makefile

```diff
@@ -11,52 +12,59 @@ favicon:
 website:
 	LONG_CACHE=false npm run build
 
-# `website` is needed for the server deploys.
-deploy: website deploy-s0 deploy-s1 deploy-s2 deploy-gh-pages deploy-gh-pages-clean
+deploy: deploy-s0 deploy-s1 deploy-s2 clean-server-deploy deploy-gh-pages deploy-gh-pages-clean
 
-deploy-s0:
+deploy-s0: prepare-server-deploy push-s0
+deploy-s1: prepare-server-deploy push-s1
+deploy-s2: prepare-server-deploy push-s2
+
+prepare-server-deploy: website
 	# Ship a copy of the front end to each server for debugging.
 	# https://github.com/badges/shields/issues/1220
-	git add -f Verdana.ttf private/secret.json build/
-	git commit -m'MUST NOT BE ON GITHUB'
-	git push -f s0 HEAD:master
-	git reset HEAD~1
-	git checkout master
+	rm -rf ${SERVER_TMP}
+	git worktree prune
+	git worktree add -B production ${SERVER_TMP}
```

Comment:
The use of git worktree feels a bit like tempting the devil. The use of `-B` in particular risks losing information without realizing.

 In general, I prefer having deployment procedures so dumb that it takes a lot for them to go wrong, and when they do, they are dumb enough that it is simple to fix them.

Reply:
#1941 is another issue with the current scripts. `git worktree` has been working more reliably for `deploy-gh-pages` than the current deploy is for the servers.